govuk_chat_evaluation/rag_answers/data_models.py

```diff
@@ -10,11 +33,108 @@ class GenerateInput(BaseModel):
 
 class EvaluationTestCase(GenerateInput):
     llm_answer: str
-    # TODO: lots more data fields
+    retrieved_context: list[str]
+    # TODO: lots more data fields [Alessia: I do not think we need extra fields here]
 
     def to_llm_test_case(self) -> LLMTestCase:
         return LLMTestCase(
             input=self.question,
+            name = str(uuid.uuid4()),
             expected_output=self.ideal_answer,
             actual_output=self.llm_answer,
+            retrieval_context=self.retrieved_context,  # type: ignore
         )
+    
+    @model_validator(mode="before")
+    @classmethod
+    def flatten_retrieved_context(cls, data: dict[str, Any]) -> dict[str, Any]:
```

Comment:
I think rather than flattening this on input, it'd be a bit easier logically if retrieved_context was a list of StructuredContexts

Then on the StructuredContext model have a method of retrieved_context (or even __str__) which does the conversation.